stdlib/public/Concurrency/Task.cpp

```diff
@@ -1624,5 +1624,13 @@ static void swift_task_asyncMainDrainQueueImpl() {
 #endif
 }
 
+SWIFT_CC(swift)
+void swift::swift_task_startOnMainActor(Job* job) {
+  AsyncTask * originalTask = swift_task_getCurrent();
+  ExecutorRef mainExecutor = swift_task_getMainExecutor();
+  swift_job_run(job, mainExecutor);
+  _swift_task_setCurrent(originalTask);
```

Comment:
Shouldn't this runtime function take in a task and convert it to a job?

Reply:
Sure, fixed the function decl. `AsyncTask` is a `Job`, and `convertTaskToJob` looks like it's just a bitcast so I don't think I need to do anything more there.